modules/message-lane/src/inbound_lane.rs

```diff
@@ -61,14 +62,25 @@ impl<S: InboundLaneStorage> InboundLane<S> {
 		}
 
 		data.latest_confirmed_nonce = outbound_lane_data.latest_received_nonce;
+		// Firstly, remove all of the records where higher nonce <= new confirmed nonce
 		while data
 			.relayers
 			.front()
-			.map(|(nonce, _)| *nonce <= data.latest_confirmed_nonce)
+			.map(|(_, nonce_high, _)| *nonce_high <= data.latest_confirmed_nonce)
 			.unwrap_or(false)
 		{
 			data.relayers.pop_front();
 		}
+		// Secondly, update the next record with lower nonce equal to new confirmed nonce if needed.
+		// Note: There will be max. 1 record to update as we don't allow messages from relayers to overlap.
+		if data
+			.relayers
+			.front()
+			.map(|(nonce_low, _, _)| *nonce_low <= data.latest_confirmed_nonce)
+			.unwrap_or(false)
```

Comment:
I see this more as a `filter()` than a `map()` operation, so what do you think about doing `filter().is_some()` instead? More legible, or nah? I know the original code had the `map().unwrap_or(false)` pattern, so don't worry about changing it, just curious

Reply:
I feel like how it was done before as it expresses how the deque is organised - by oldest messages at the front thus we start from the front the pop only until the set is updated. `filter()` would go over all of the values ignoring the fact they're ordered, so maybe let's keep it as it is?

Reply:
Well we're performing both the `map()` and `filter()` calls on an `Option` after calling `front()`, so we wouldn't be going through all the values.

Again, I was just curious as to what you thought, keep the code as is